common.ts

```diff
@@ -180,15 +180,13 @@ export async function attemptHttpRequest<B, R>(
 /**
  * creates an axios request function
  */
-export const createRequestFunction = function (axiosArgs: RequestArgs, axiosInstance: AxiosInstance, configuration: Configuration, credentials?: Credentials) {
```

Comment:
Customers can function to extend the supported methods while still benefiting from error handling, retries and auth. We used this for the Playground when the SDK dropped support for ReadTuples, but users who were on old versions of OpenFGA still needed, and it can also be used to target future API methods without bumping the SDKs.

I would be OK changing it, but maybe a better change would be to move https://github.com/openfga/js-sdk/blob/main/common.ts#L183-L205 to `api.ts` as that is the only place it is used, and then export it from there? or even have it as a static public method on baseAPI https://github.com/openfga/js-sdk/blob/main/base.ts#L39C14-L39C21

Reply:
I think moving it to be a method on the class would make sense but I think realistically we should try to do that in a cross-SDK way by like providing a `request` function like below onto all clients to allow developers to make custom requests as they need rather than a one-off change in this SDK.

```js
await fgaclient.request(urlPath, { method: "POST", data, headers });
```

Reply:
Yeah that would be a good idea to have, we can add it as part of the SDKs 1.0 work